File: capacity-club-api/src/domain-modules/address/model/payload/address-update.payload.ts
import { ApiCodeResponse } from '@common/api';
import { addressProperties } from '@common/documentation';
import { UniqueId } from '@common/model/unique-id';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsOptional, IsString, Length } from 'class-validator';

/**
 * Class used to update address.
 * This class defines the payload used when updating existing address.
 */
export class AddressUpdatePayload {
  @ApiProperty({
    name: addressProperties.address_id.name,
    description: addressProperties.address_id.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_ID_IS_MISSING,
  })
  @Length(26, 26)
  address_id: UniqueId;
  @ApiProperty({
    name: addressProperties.street.name,
    description: addressProperties.street.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_STREET_IS_MISSING,
  })
  @IsString()
  street: string;
  @ApiProperty({
    name: addressProperties.number.name,
    description: addressProperties.number.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_NUMBER_IS_MISSING,
  })
  @IsString()
  number: string;
  @ApiProperty({
    name: addressProperties.zipcode.name,
    description: addressProperties.zipcode.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_ZIPCODE_IS_MISSING,
  })
  @IsString()
  zipcode: string;
  @ApiProperty({
    name: addressProperties.town.name,
    description: addressProperties.town.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_TOWN_IS_MISSING,
  })
  @IsString()
  town: string;
  @ApiProperty({
    name: addressProperties.country.name,
    description: addressProperties.country.description,
  })
  @IsNotEmpty({
    message: ApiCodeResponse.ADDRESS_PAYLOAD_COUNTRY_IS_MISSING,
  })
  @IsString()
  country: string;
  @ApiProperty({
    name: addressProperties.complement.name,
    description: addressProperties.complement.description,
  })
  @IsOptional()
  @IsString()
  complement: string;
}

File: capacity_club_mobile_app/lib/domain-features/data/address/model/payload/address_update_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:equatable/equatable.dart';

class AddressUpdatePayload extends Equatable implements Payload {
  final UniqueId address_id;
  final String street;
  final String number;
  final String zipcode;
  final String town;
  final String country;
  final String complement;

  AddressUpdatePayload({
    required this.address_id,
    required this.street,
    required this.number,
    required this.zipcode,
    required this.town,
    required this.country,
    this.complement = '',
  });

  Map<String, dynamic> toJson() => {
        'address_id': address_id,
        'street': street,
        'number': number,
        'zipcode': zipcode,
        'town': town,
        'country': country,
        'complement': complement,
      };

  @override
  List<Object?> get props =>
      [address_id, street, number, zipcode, town, country, complement];
}
